### PR TITLE
Bug 1372724 - Add support for logging error events to Sentry

### PR DESCRIPTION
**Pull request by justindarc**

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
AppDelegate.swift | 38 | Force casts should be avoided. 
AppDelegate.swift | 205 | Force casts should be avoided. 
AppDelegate.swift | 214 | if,for,while,do statements shouldn't wrap their conditionals in parentheses. 
AppDelegate.swift | 247 | if,for,while,do statements shouldn't wrap their conditionals in parentheses. 
AppDelegate.swift | 207 | Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. 
AppDelegate.swift | 229 | Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. 
AppDelegate.swift | 144 | When registing for a notification using a block, the opaque observer that is returned should be stored so it can be removed later. 
AppDelegate.swift | 151 | When registing for a notification using a block, the opaque observer that is returned should be stored so it can be removed later. 
BrowserDB.swift | 42 | Force tries should be avoided. 
BrowserDB.swift | 88 | Force tries should be avoided. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Shared/SentryIntegration.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    }
+
+    public func crash() {
+        Client.shared?.crash()
+    }
+    
+    public func send(message: String, tag: String = "general", severity: SentrySeverity = .info, completion: SentryRequestFinished? = nil) {
+        if !enabled {
+            if completion != nil {
```
Comment (farhanpatel) [inline Shared/SentryIntegration.swift]: can you do a completion?(nil) here instead and not have to check if its nil?

File: Shared/SentryIntegration.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    public func sendSync(message: String, tag: String = "general", severity: SentrySeverity = .info) {
+        if !enabled {
+            return
+        }
+
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+
+        send(message: message, tag: tag, severity: severity, completion: { (error) in
+            dispatchGroup.leave()
```
Comment (farhanpatel) [inline Shared/SentryIntegration.swift]: Is this how you can log a crash before the app fully crashes?

File: Storage/SQL/BrowserDB.swift [hunk 3]
```diff
@@ -220,7 +219,9 @@ open class BrowserDB {
             }
             return success
         }) {
-            // Err getting a transaction
+            // Error getting a transaction
+            log.error("Unable to get a transaction: \(error.localizedDescription)")
+            SentryIntegration.shared.sendSync(message: error.localizedDescription, tag: "BrowserDB", severity: .error)
```
Comment (farhanpatel) [inline Storage/SQL/BrowserDB.swift]: For this specific error is there not anything else you would like to log? Or will that be in another PR?

Comment (justindarc) [inline Storage/SQL/BrowserDB.swift]: There's a lot more I want to log in this method in general. Just want to get this landed for now.

File: Shared/SentryIntegration.swift (hunk 2 above)
Comment (justindarc) [inline Shared/SentryIntegration.swift]: Yes. This is why I made a second method called `sendSync()` because most cases you wouldn't want to block on it.
